odoo/service/server.py

```diff
@@ -443,9 +443,9 @@ def worker_spawn(self, klass, workers_registry):
 
     def long_polling_spawn(self):
         nargs = stripped_sys_argv()
-        cmd = os.path.join(os.path.dirname(nargs[0]), "odoo.py")
-        nargs = [cmd, 'gevent'] + nargs[1:]
-        popen = subprocess.Popen([sys.executable] + nargs)
+        parent = os.path.dirname(config['root_path'])
+        cmd = [sys.executable, '-m', 'odoo', 'gevent'] + nargs[1:]
```

Comment:
@sle-odoo in some cases `odoo --workers` fails because it does not find `__main__.py`.

I made this as an alternative: https://github.com/odoo-dev/odoo/pull/108